File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string compare = strs[0];
        if(compare==""){
            return "";
        }
        int res = INT_MAX;
        for(int i=1;i<strs.size();i++){
            int temp = 0;
            string compare2 = strs[i];
            int j = 0;
            while(j<compare.size() && compare[j]==compare2[j]){
                temp++;
                j++;
            }
            res = min(res, temp);
        }

        return compare.substr(0, res);
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        
        size = len(strs)
        res = 1000
        for i in range (1, size):
            count = 0
            current = strs[i]
            prev = strs[i-1]
            index = 0
            while index < len(current) and index < len(prev) and index < res:
                if (current[index] == prev[index]):
                    count += 1
                else:
                    break
                index += 1
            res = min(res, count)
        
        return strs[0][0:res]
